tests/Cluster.py

```diff
@@ -245,9 +245,29 @@ def insertSpecificExtraNodeosArgs(node, insertStr):
 
             certAuth = os.path.join(privacyDir, "CA_cert.pem")
             def getArguments(number):
-                nodeCert = os.path.join(privacyDir, "node{}.crt".format(number))
-                nodeKey = os.path.join(privacyDir, "node{}_key.pem".format(number))
-                return "--p2p-tls-own-certificate-file {} --p2p-tls-private-key-file {} --p2p-tls-ca-certificate-file {}".format(nodeCert, nodeKey, certAuth)
+                # this function converts number to eos name string
+                # eos name can have only numbers 1-5
+                # e.g. 0 -> 1, 6 -> 5.1, 12 -> 5.5.2
+                def normalizeNumber(number):
+                    if number <= 0:
+                        number = 1
```

Comment:
won't this method return "1" for number=1 and number=0? I think this should be
max = 4
if number >= 0 and number <= max:
    return str(number+1)
 ...
       cnt -= max
       ...

Reply:
logic of this method is to normalize number so I think it shouldn't increase it. However your concern with 0 and 1 makes sense, it is may result in bug if caller will start from zero. What about forbidding zero at all? just putting an `assert(number > 0)`

Reply:
well this is called with the node number that starts with 0, so it has to take 0, or else we need to adjust the number we are passing in, but that makes no sense to me, just do it here.  We also need to replicate this logic in the generate certificates script, creating node1,node2,...node5,node1.1,...

Reply:
and the while would be:

while cnt > max:

Reply:
actually, you are correct, when getArgument calls getNumber, it should add 1 to the passed in number and it should treat 1 as "1", 2 as "2", etc. So yes, assert on num==0